### PR TITLE
Ray 24 add jcr

### PR DESCRIPTION
**Pull request by gesparza3**
This PR creates adds modules for creating an instance of Artifactory-JCR for publishing OCI compliant artifacts. The included modules will create an instance of JCR in the `apps` stage and create Docker & Helm repos within the `config` stage. Each new instantiation of a product will create a new _user_ in Jenkins to allow pipelines to publish to Artifactory.

### CODE REVIEW

**Review by RobertKelly — APPROVED** [review]
Looks good!

File: stages/apps/lead/variables.tf [hunk 1]
```diff
@@ -75,6 +75,10 @@ variable "enable_harbor" {
   default = true
 }
 
+variable "enable_artifactory_jcr" {
+  default = true
```
Comment (mrparkers) [inline stages/apps/lead/variables.tf:79]: I don't think this should be `true`.  lead-terraform's primary use case is still for our lab purposes, so we don't want to accidentally leave this on for a lab.  I'd prefer this be `false` and we can explicitly override this in lead-environments.

File: modules/lead/product-jenkins/artifactory.tf [hunk 2]
```diff
@@ -1,34 +1,12 @@
...
-  data = {
-    "settings.xml" = data.template_file.maven_settings[0].rendered
-  }
-
-  type = "Opaque"
+resource "artifactory_user" "robot" {
+  count      = var.enable_artifactory ? 1 : 0
+  name       = "robot-${var.product_name}" 
+  email      = "robot-${var.product_name}@liatrio.com"
+  password   = data.random_password.robot_password.result
```
Comment (mrparkers) [inline modules/lead/product-jenkins/artifactory.tf:11]: is there a reason these changes are within the `artifactory.tf` file and not a new file?  is the plan to completely deprecate artifactory usage within lead-terraform?

**Review by mrparkers — CHANGES_REQUESTED** [review]

File: modules/lead/product-jenkins/artifactory.tf (hunk 2 above)
Comment (RobertKelly) [reply modules/lead/product-jenkins/artifactory.tf:11]: Maybe?

Comment (mrparkers) [reply modules/lead/product-jenkins/artifactory.tf:11]: if so, I'd still prefer changes like that to show up in another PR, and not do some of them in this one, then follow up with the rest.

can we add these changes in a new file?  also, how do these user credentials get seeded to jenkins?

Comment (gesparza3) [reply modules/lead/product-jenkins/artifactory.tf:11]: Artifactory already is deprecated or it was back when we did the big refactor. This was a leftover file that I repurposed.

File: stages/apps/lead/artifactory.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+data "vault_generic_secret" "artifactory" {
+  path = "lead/aws/${data.aws_caller_identity.current.account_id}/artifactory"
```
[the remark this answers is not in the document]
Comment (gesparza3) [reply stages/apps/lead/artifactory.tf:2]: No not yet, I don't have access on my end to do that. Figured we'd resolve any comments on this PR running the terraform against our cluster.

File: modules/lead/product-jenkins/artifactory.tf (hunk 2 above)
Comment (mrparkers) [reply modules/lead/product-jenkins/artifactory.tf:11]: gotcha, can we rename the file then so it's clear that this is specifically for JCR?

File: stages/apps/lead/artifactory.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrparkers) [reply stages/apps/lead/artifactory.tf:2]: @dogbern should have access.  I'll open a PR now to get you access too

File: stages/config/lead/variables.tf [hunk 4]
```diff
@@ -18,6 +18,8 @@ variable "enable_keycloak" {}
 
 variable "enable_harbor" {}
 
+variable "enable_artifactory_jcr" {}
```
Comment (mrparkers) [inline stages/config/lead/variables.tf:21]: let's add `default = false` here

File: stages/config/lead/variables.tf [hunk 5]
```diff
@@ -26,6 +28,10 @@ variable "harbor_hostname" {
   description = "Harbor hostname provided by app stage"
 }
 
+variable "artifactory_jcr_hostname" {
+  description = "Artifactory hostname provided by app stage"
```
Comment (mrparkers) [inline stages/config/lead/variables.tf:32]: let's add a default here too so we don't have to provide this if artifactory isn't enabled

File: stages/apps/lead/variables.tf (hunk 1 above)
Comment (mrparkers) [reply stages/apps/lead/variables.tf:79]: still looking to get this resolved

**Review by mrparkers — APPROVED** [review]

**Pull request merged**